rust/yarp/build.rs

```diff
@@ -0,0 +1,507 @@
+                    writeln!(file, "            None")?;
+                    writeln!(file, "        }} else {{")?;
+                    writeln!(file, "            Some(Location(location))")?;
+                    writeln!(file, "        }}")?;
+                    writeln!(file, "    }}")?;
+                },
+                NodeParamType::LocationList => {
+                    writeln!(file, "    pub fn {}(&self) -> LocationList {{", node_param.name)?;
+                    writeln!(file, "        let list: *mut yp_location_list_t = unsafe {{ &mut (*self.0).{} }};", node_param.name)?;
+                    writeln!(file, "        LocationList(list)")?;
+                    writeln!(file, "    }}")?;
+                },
+                NodeParamType::UInt32 => {
+                    writeln!(file, "    pub fn {}(&self) -> u32 {{", node_param.name)?;
+                    writeln!(file, "        unsafe {{ (*self.0).{} }}", node_param.name)?;
+                    writeln!(file, "    }}")?;
+                },
+                NodeParamType::Flags => {
+                    writeln!(file, "    pub fn {}(&self) -> yp_node_flags_t {{", node_param.name)?;
+                    writeln!(file, "        unsafe {{ (*self.0).base.flags }}")?;
+                    writeln!(file, "    }}")?;
+                }
+            }
+
+            padding = true;
+        }
+    }
+
+    writeln!(file, "}}")?;
+    writeln!(file)?;
+
+    writeln!(file, "impl std::fmt::Debug for {} {{", node.name)?;
+    writeln!(file, "    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {{")?;
+
+    write!(file, "        write!(f, \"{}(", node.name)?;
+    if let Some(child_nodes) = &node.child_nodes {
+        let mut padding = false;
+        for _ in child_nodes {
+            if padding { write!(file, ", ")?; }
+            write!(file, "{{:?}}")?;
+            padding = true;
+        }
+
+        write!(file, ")\", ")?;
+        padding = false;
+
+        for node_param in child_nodes {
+            if padding { write!(file, ", ")?; }
+            write!(file, "self.{}()", node_param.name)?;
+            padding = true;
+        }
+    } else {
+        write!(file, ")\"")?;
+    }
+
+    writeln!(file, ")")?;
+    writeln!(file, "    }}")?;
+    writeln!(file, "}}")?;
+
+    Ok(())
+}
+
+/// Write the bindings to the `$OUT_DIR/bindings.rs` file. We'll pull these into
+/// the actual library in `src/lib.rs`.
+fn write_bindings(config: Config) -> Result<(), Box<dyn std::error::Error>> {
+    let out_path = PathBuf::from(std::env::var_os("OUT_DIR").unwrap()).join("bindings.rs");
+    let mut file = std::fs::File::create(&out_path).expect("Unable to create file");
+
+    writeln!(file, "#[allow(clippy::wildcard_imports)]")?;
+    writeln!(file, "use yarp_sys::*;")?;
+    writeln!(file)?;
+
+    writeln!(file, "/// A range in the source file.")?;
+    writeln!(file, "#[derive(Debug)]")?;
+    writeln!(file, "pub struct Location(*mut yp_location_t);")?;
```

Comment:
Is the intent here (and maybe with other types as well?) to implement accessor type methods later? When I look at the docs for this (which can be helpful while you're in this stage of development; just run `cargo doc`), it doesn't look like there's any way to get the start and end points of the Location.

Reply:
I've implemented the accessor methods now!

The main purpose of this PR is just to get something working that people can iterate on. 🤷